### ISSUE TITLE
More dupe glitches

### ISSUE THREAD

**Issue by JamesClarke7283** [ketwaroo/minetest-k-recycler#4]
If you put 5 diamonds i in the recycler, it produces a diamond block.

If you put 5 raw gold or raw iron in, it produces a raw gold/iron block. same for raw copper and redstone dust.

Which lets people dupe it.

Also with the spawncraft mod installed, you can uncraft a villager egg and get unlimited emeralds. and not just villager spawn eggs, but all of them produce self replication.

I fixed it in my fork, but your solution is better. you may need to make the no self replicating items, smarter.

**Comment by ketwaroo** [comment]
It doesn't have anything to do with self replicating items.

It's to do with .. 

```
# Minimum stack ratio to allow partial recycling.
# 0.0 disables it and any extra items that can't be recycled will just pass through or get discarded if there's no space on the output grid.
# 1.0 always try partial recycling. May cause free stuff.
k_recyclebin.partial_recycling_minimum_ratio (Partial Recycling Minimum stack Ratio) float 0.5 0.0 1.0
```

because that particular recipe is 9 diamonds -> one block -> 9 diamonds, the min stack of 4.5 is satisfied. Originally this was meant to deal with left over stairs stacks.

will adjust in a bit.

**Comment by ketwaroo** [comment]
fixed in 198361b

if the recycler output is a single item, the full input stack size is required.
